client/notices.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) 2024 Canonical Ltd
```

Comment:
I'm commenting on the first line that this patch should be split.

The changes to client are stand-alone and can be proposed straight way.

The changes to cmd/snap should be split out and can depend on the larger patch-set.

Reply:
Updated, Thank you!